ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineGetBlobsV1.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * <p>3. Client software **MUST** support request sizes of at least 128 blob versioned hashes. The
+ * client **MUST** return `-38004: Too large request` error if the number of requested blobs is too
+ * large.
+ *
+ * <p>4. Client software **MAY** return an array of all `null` entries if syncing or otherwise
+ * unable to serve blob pool data.
+ *
+ * <p>5. Callers **MUST** consider that execution layer clients may prune old blobs from their pool,
+ * and will respond with `null` if a blob has been pruned.
+ */
+public class EngineGetBlobsV1 extends ExecutionEngineJsonRpcMethod {
+
+  private final Map<VersionedHash, BlobsWithCommitments.BlobQuad> blobMap = new HashMap<>();
+  private final BlobCache blobCache;
+
+  public EngineGetBlobsV1(
+      final Vertx vertx,
+      final ProtocolContext protocolContext,
+      final EngineCallListener engineCallListener,
+      final TransactionPool transactionPool) {
+    super(vertx, protocolContext, engineCallListener);
+    this.blobCache = transactionPool.getBlobCache();
+    transactionPool.subscribePendingTransactions(this::onTransactionAdded);
+    transactionPool.subscribeDroppedTransactions(this::onTransactionDropped);
+  }
+
+  @Override
+  public String getName() {
+    return "engine_getBlobsV1";
+  }
+
+  @Override
+  public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext) {
+    final VersionedHash[] versionedHashes;
+    try {
+      versionedHashes = requestContext.getRequiredParameter(0, VersionedHash[].class);
+    } catch (JsonRpcParameter.JsonRpcParameterException e) {
+      throw new InvalidJsonRpcParameters(
+          "Invalid versioned hashes parameter (index 0)",
+          RpcErrorType.INVALID_VERSIONED_HASHES_PARAMS,
+          e);
+    }
+
+    if (versionedHashes.length > 128) {
+      return new JsonRpcErrorResponse(
+          requestContext.getRequest().getId(),
+          RpcErrorType.INVALID_ENGINE_GET_BLOBS_V1_TOO_LARGE_REQUEST);
+    }
+
+    final List<BlobAndProofV1> result = getResult(versionedHashes);
+
+    return new JsonRpcSuccessResponse(requestContext.getRequest().getId(), result);
+  }
+
+  private @Nonnull List<BlobAndProofV1> getResult(final VersionedHash[] versionedHashes) {
+    return Arrays.stream(versionedHashes)
+        .map(this::getBlobQuad)
+        .map(this::getBlobAndProofV1)
+        .toList();
+  }
+
+  private @Nullable BlobAndProofV1 getBlobAndProofV1(final BlobsWithCommitments.BlobQuad bq) {
+    if (bq == null) {
+      return null;
+    }
+    return new BlobAndProofV1(
+        bq.blob().getData().toHexString(), bq.kzgProof().getData().toHexString());
+  }
+
+  private BlobsWithCommitments.BlobQuad getBlobQuad(final VersionedHash vh) {
+
+    BlobsWithCommitments.BlobQuad blobQuad = blobMap.get(vh);
+    if (blobQuad == null) {
+      blobQuad = blobCache.get(vh);
+    }
+
+    return blobQuad;
+  }
+
+  public void onTransactionAdded(final Transaction transaction) {
```

Comment:
idk if the RPC should be itself managing the blob cache?

Reply:
I was wondering where that should live. The other option would be the TransactionPool. The BlobCache is taken care of in the TransactionPool, because these blobs are needed in case we have a reorg.
The blobMap is not needed by the TransactionPool. Here we just keep track of the blobs that we have available in our transaction pool and we have to update the map when blob transactions are added or removed from the pool.

Reply:
I have moved that logic into the TransactionPool